### ISSUE TITLE
Landmarks issues for Accessibility

### ISSUE THREAD

**Issue by 00travelgirl00** [WordPress/twentytwentyfour#74]
I have made a look at the landmarks and noticed following problems. See also handbook regarding landmarks: https://make.wordpress.org/themes/handbook/review/accessibility/required/#aria-landmark-roles 
 
homepage:
- main landmark is not present.
- not all content is inside a landmark 

pages:
- Titel / H1 is not in main

posts:
- Titel / H1 is not in main
- meta informations and comments are not part of any landmark. main landmark would be good

## suggestion
archive:
- wrap posts in `<article>` instead of divs

I'm not sure, if this last point is a requirement. But I have seen this practice with `<article>` several times and I think its a nice practice and user friendly for screen-readers.

**Comment by graylaurenm** [comment]
I've worked on this for the homepage, pages, and archives. I haven't had a change to look at the posts yet. I will submit a pull request momentarily.

**Comment by graylaurenm** [comment]
I noticed that `home-portfolio` and `home-writer` also need landmarks.

**Comment by richtabor** [comment]
How’re we looking on this front? Did we over all the bases?

**Comment by carolinan** [comment]
All templates are not complete yet and here are still problems.

Header: It was decided that the sticky header should be used everywhere. I think the wrapping group that makes the template part sticky should be the `<header>` element.
Current:
```
<div>
  <header>
  stuff
  </header>
</div>
<main></main>
<footer></footer>

```
Expected:

```
<header>
  <div>
  stuff
  </div>
</header>
<main></main>
<footer></footer>
```


404,, archive, home.html, index,  page-alternative, page-no-title, page-portfolio, page
Main landmark is present.
All content is inside a landmark

page-writer: For some reason the `<main> `is inside another group that uses `<div>.` These could be switched so that the `<main>` is the first group after the header. Let's make this consistent.

Search: has content outside landmarks. Fixed by #187

single-with-sidebar: Has content outside landmarks, missing landmark for the sidebar.

single-writer: Has content outside landmarks
single: Has content outside landmarks

home-writer: Has not been re-created using patterns yet and is missing landmarks.

**Comment by MaggieCabrera** [comment]
yeah, the template work is still on going, in fact the header PR hasn't been merged yet, but this is a good recap to have, thank you

**Comment by MaggieCabrera** [comment]
I think most of the templates are in now (only the portfolio ones are a WIP), let's have another pass and make sure we are covering all the landmarks

**Comment by MaggieCabrera** [comment]
I went over this and I think all landmarks are present
